pkg/resource/deploy/source_eval.go

```diff
@@ -2261,6 +2370,39 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 
 	additionalSecretOutputs := opts.GetAdditionalSecretOutputs()
```

Comment:
hooks need adding to TransformResourceOptions and flowing through transforms above this